docs/src/content/en/guides/getting-started/vite-react.mdx

```diff
@@ -8,248 +8,422 @@ import TabItem from "@theme/TabItem";
 
 # Integrate Mastra in your React + Vite project
 
-Mastra integrates with Vite, making it easy to:
+In this guide, you'll build a tool-calling AI agent using Mastra, then connect it to React by calling the agent directly from Mastra's standalone server.
 
-- Build flexible APIs to serve AI-powered features
-- Simplify deployment with a unified codebase for frontend and backend
-- Take advantage of Mastra's Client SDK
+You'll use [AI SDK UI](https://ai-sdk.dev/docs/ai-sdk-ui/overview) and [AI Elements](https://ai-sdk.dev/elements) to create a beautiful, interactive chat experience.
 
-Use this guide to scaffold and integrate Mastra with your React + Vite project.
+## Before you begin
 
-:::warning
+- You'll need an API key from a supported [model provider](/models/v1). If you don't have a preference, use [OpenAI](/models/v1/providers/openai).
+- Install Node.js `v22.13.0` or later
 
-This guide assumes you're using React + Vite with React Router v7 at the root of your project, e.g., `app`.
+## Create a new React + Vite app (optional)
```

Comment:
Did you consider a boilerplate template, like you did for Express?

Reply:
That was only an escape-hatch, I don't really want to maintain those repositories 😅 

If we wouldn't use AI Elements here we wouldn't need the Tailwind setup